react-hot-button-component/my-app/src/App.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { useState, useEffect } from 'react';
+
+function App() {
+  const [count, setCount] = useState(0);
+  const [currentColor, setCurrentColor] = useState('cold');
+  // allows for state to automatically update the count before the loop ends ?
+
+  useEffect(() => {
```

Comment:
We haven't learned `useEffect` yet. You won't need it in any of the exercises we do until we have an exercise that covers it.

As it turns out, this isn't a very good use of `useEffect`. You usually only use `useEffect` when you are interacting with non-React things, such as the network.